perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -2234,9 +2234,14 @@ public User convertUserEmptyStringsInObjectAttributesIntoNull(User user) {
 	public void changeNonAuthzPassword(PerunSession sess, User user, String m, String password, String lang) throws InternalErrorException, LoginNotExistsException, PasswordChangeFailedException, PasswordOperationTimeoutException, PasswordStrengthFailedException {
 
 		String requestId = Utils.cipherInput(m, true);
-		String namespace = getUsersManagerImpl().loadPasswordResetRequest(user, Integer.parseInt(requestId));
+		Pair<String,String> resetRequest = getUsersManagerImpl().loadPasswordResetRequest(user, Integer.parseInt(requestId));
 
-		if (namespace.isEmpty()) throw new InternalErrorException("Password reset request is not valid anymore or doesn't existed at all for User: "+user);
+		if (resetRequest == null) {
+			throw new InternalErrorException("Password reset request is not valid anymore or doesn't existed at all for User: "+user);
+		}
+
+		String namespace = resetRequest.getLeft();
+		String mail = resetRequest.getRight();
```

Comment:
You still can get null in left or right part. Shouldn't we check it here too?

Reply:
Namespace is non-null in DB. Mail can be null and I added check before adding its value between known mails at line 2268.